### ISSUE TITLE
Federation: Multiple announce activities of same toot from same user is ignored.

### ISSUE THREAD

**Issue by chocological00** [tootsuite/mastodon#13767]
Some AP implementations (ex. Misskey) allows boosting the same toot multiple times.
The W3C [doesn't mention](https://www.w3.org/TR/2018/REC-activitypub-20180123/#announce-activity-inbox) about multiple announce activity should not be done, so I am assuming that this is not a non-standard feature. Please correct me if I'm wrong.

Imagine a situation:
1. User from instance A (not Mastodon) boosts a toot in Jan 2020.
2. Users in Instance B (Mastodon) can see the boosted toot.
3. Same user in instance A finds the same toot relevant to the discussion in May 2020. S/he boosts the same toot.

### Expected behaviour
4-1. Users in instance B can see the boosted toot.

### Actual behaviour
4-2. Users in instance B cannot see the boosted toot.
(Seems like Mastodon ignores duplicate announce activities)

### Specifications
Misskey v11.37.0 (Latest) -> Mastodon v3.1.4

**Comment by ThibG** [comment]
Yeah I'm not completely sure why we don't allow boosting the same thing multiple times.
I guess if we don't change that, from a federation standpoint, what we could do instead is *replace* the old boost with the new one.
